core/src/main/java/com/netease/arctic/utils/map/SimpleSpillableMap.java

```diff
@@ -111,6 +111,11 @@ public void put(K key, T value) {
       this.currentInMemoryMapSize = this.memoryMap.size() * this.estimatedPayloadSize;
     }
 
+    if (memoryMap.containsKey(key)) {
```

Comment:
Maybe we can put the new logic in the else of the following condition (this.currentInMemoryMapSize < maxInMemorySizeInBytes) to reduce calculations.